src/test/java/array/Array1dTest.java

```diff
@@ -0,0 +1,207 @@
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldCreateEmptyArrayByFactoryMethod() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = Array1d.of(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldVisitEmptyArray() {
+        // given
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() -> array.visit(i -> System.out.print(i + ", ")));
+    }
+
+    @Test
+    void shouldFillArrayWithGivenNumbers() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertDoesNotThrow(() ->
+                IntStream.range(start, end)
+                        .forEach(i -> array.insert(
+                                Coordinates1d.of(i),
+                                10 * i)));
+    }
+
+    @Test
+    @DisplayName("Should throw an exception at second insertion into already occupied cell")
+    void shouldThrowAnException() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates1d.of(i),
+                        10 * i));
+        // expect
+        assertThrows(IllegalArgumentException.class,
+                () -> array.insert(Coordinates1d.of(1), 100));
+    }
+
+    @Test
+    @DisplayName("Should throw an exception if index out of bound")
+    void shouldThrowAnExceptionAlso() {
+        // given
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        // expect
+        assertThrows(ArrayIndexOutOfBoundsException.class,
+                () -> array.insert(Coordinates1d.of(2 * size), 120));
+    }
+
+
+    @Test
+    void shouldVisitNonEmptyArray() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates1d.of(i),
+                        10 * i));
+        // expect
+        assertDoesNotThrow(() -> array.visit(i -> System.out.print(i + ", ")));
+    }
+
+    @ParameterizedTest
+    @CsvSource({"1, 10", "5, 50"})
+    void shouldGiveCellValueByIndex(Integer index, Integer expectedValue) {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array1d<Integer> array = Array1d.of(size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates1d.of(i),
+                        10 * i));
+        // when
+        Optional<Integer> value = array.at(Coordinates1d.of(index));
+        // then
+        value.ifPresent(
```

Comment:
it should always be present, so
`value.map(...).orElseThrow()`
should be used, IMHO

Reply:
No, it shouldn't. If you don't put anything by some `index` this cell will be empty and `at` method will return `Optional.empty()` then.

Reply:
Why?
fo this input
```
@CsvSource({"1, 10", "5, 50"})
```
always should

And if you want to check the scenario when you pass illegal index, it should be either covered by separate test, or input should be like
```
index,
value,
isValueShouldBeFoundForIndex
```

For example if you pass `"10, 100"` test will faill with exception `OutOfBoundsException` but in all other test you check this with `assertDoesNotThrow` but this looks a bit inconsistent.

If you're ok to keep it this way, ok I will not insist.

Reply:
up

Reply:
Please compare and choose
`        value.map(v -> {
            assertEquals(expectedValue, v);
            return v;
        }).orElseThrow();`

vs.

`        value.ifPresent(
                v -> assertEquals(expectedValue, v));`

Reply:
Am I missing smth?

the second option is the current state of affairs. 
If you don't like the idea of validating that value should be present , ok - I will not insist.

Reply:
the second option is the current state of affairs. - yes, it's a current state of code.

In the first approach I don't like `return v;` part. It's sophisticated and hard to read. Hense won't fix. If agree - please close

Reply:
```suggestion
       val value = array.at(Coordinates1d.of(index)).orElseThrow(new RuntimeException("Value should be present"));
       assertEquals(expectedValue, value));
        
```

Reply:
Fixed 014da09b2f32677c9fd5690d11cd3210f0d08ab1 but without exception's text message